common/src/hash.rs

```diff
@@ -53,6 +53,15 @@ impl HashSecret {
         fix_sentinel(mod_int(self.hash_one(data) as _))
     }
 
+    pub fn hash_integer<T>(&self, data: T) -> PyHash
+    where
+        T: Into<BigInt>,
+    {
+        let bigint = data.into();
```

Comment:
The overall approach is good, but do not create a BigInt object. The hash function need to be refactored to share usize implementation

Reply:
I see. Fixed